lib/sequenceserver/blast/formatter.rb

```diff
@@ -38,11 +39,15 @@ def run
         command =
           "blast_formatter -archive '#{archive_file}'" \
           " -outfmt '#{format} #{specifiers}'" \
-          " -out '#{file.path}' 2> /dev/null"
-        logger.debug("Executing: #{command}")
-        Dir.chdir(File.exist?(DOTDIR) && DOTDIR || Dir.pwd) do
-          system(command)
-        end
+          " -out '#{file}' 2> /dev/null"
+          if File.exist?(file)
```

Comment:
I would have placed the check the first thing in run method, logged and returned if file existed, and left the rest untouched.